library/core/class.format.php

```diff
@@ -1713,6 +1714,49 @@ public static function mentions($Mixed) {
         }
     }
 
+    /**
+     * Optionally replace any embedded images from untrusted sites with links to the image.
+     *
+     * @param string $html String from which you might want to strip out untrusted images.
+     * @return string The string with images replaced by links.
+     */
+    public static function sanitizeImages($html) {
+        // If the config does not have a SanitizeImages entry or it is set to false, pass the HTML to the renderer as is.
+        if (!c('Garden.Format.SanitizeImages', false)) {
+            return $html;
+        }
+
+        preg_match_all('/\<img\s+src\s*=\s*[\"\'](.*?)[\"\'].*\>/', $html, $matches, PREG_SET_ORDER);
+        // If no images are found in the HTML, pass the HTML to the renderer as is.
+        if (!$matches) {
+            return $html;
+        }
+
+        foreach ($matches as $image)  {
+            $imageUrl = (isset($image[1])) ? parse_url($image[1]) : null;
+            $stripImage = true;
+
+            // If the image is coming from this forum's CDN, do not strip this image.
+            if (c('S3.Prefix') && c('S3.Zone')) {
```

Comment:
Gdn_Format should not know about this. My suggestion is that you make the S3 plugin add a trusted domain (although see my note below).